FWCore/ParameterSet/python/Config.py

```diff
@@ -1431,6 +1442,37 @@ def validate(self):
         #    raise RuntimeError("No input source was found for this process")
         pass
 
+    def handleProcessAccelerators(self, parameterSet):
+        # Sanity check
+        useSet = set(self.options.accelerators.value())
+        accSet = set([label for acc in self.__dict__['_Process__accelerators'].values() for label in acc.labels()])
+        accSet.add("auto")
+        diff = useSet.difference(accSet)
+        if len(diff) > 0:
+            invalid = ",".join(diff)
+            valid = ",".join(sorted(list(accSet)))
+            raise ValueError("Invalid value{} of {} in process.options.accelerators, valid values are {}".format("s" if len(diff) > 2 else "",
+                                                                                                                 invalid,
+                                                                                                                 valid))
+
+        availableAccelerators = set()
+        for acc in self.__dict__['_Process__accelerators'].values():
+            for l in acc.enabledLabels():
+                availableAccelerators.add(l)
+        availableAccelerators = sorted(list(availableAccelerators))
+        parameterSet.addVString(False, "@available_accelerators", availableAccelerators)
```

Comment:
I originally wanted to pass the list of available accelerators via the `process.options` rather than as a "top-level parameter", but couldn't find a simple way to add it only to the "process description" without it ending up in the python counterpart of `process.options`.